examples/02_decoding/plot_miyawaki_encoding.py

```diff
@@ -0,0 +1,263 @@
+# We now stack the X and y data and remove an offset in the beginning/end.
+
+X_train = np.vstack([x[2:] for x in X_train])
+y_train = np.vstack([y[:-2] for y in y_train]).astype(float)
+
+##############################################################################
+# X_train is a matrix of *samples* x *voxels*
+
+print(X_train.shape)
+
+##############################################################################
+# We flatten the last two dimensions of y_train
+# so it is a matrix of *samples* x *pixels*.
+
+# Flatten the stimuli
+y_train = np.reshape(y_train, (-1, y_shape[0] * y_shape[1]))
+
+print(y_train.shape)
+
+##############################################################################
+# Building the encoding models
+# ----------------------------
+# We can now proceed to build a simple **voxel-wise encoding model** using
+# `Ridge regression <http://en.wikipedia.org/wiki/Tikhonov_regularization>`_.
+# For each voxel we fit an independent regression model,
+# using the pixel-values of the visual stimuli to predict the neuronal
+# activity in this voxel.
+
+from sklearn.linear_model import Ridge
+from sklearn.cross_validation import KFold
+
+##############################################################################
+# Using 10-fold cross-validation, we partition the data into 10 'folds',
+# we hold out each fold of the data for testing, then fit a ridge regression
+# to the remaining 9/10 of the data, using y_train as predictors
+# and X_train as targets, and create predictions for the held-out 10th.
+
+estimator = Ridge(alpha=100.)
+cv = KFold(len(y_train), 10)
+
+predictions = []
+for train, test in cv:
+    predictions.append(Ridge(alpha=100.).fit(
+    y_train.reshape(-1, 100)[train], X_train[train]).predict(
+        y_train.reshape(-1, 100)[test]))
+
+##############################################################################
+# To have a measure for the quality of our encoding model, we estimate how
+# much variance the encoding model explains in each voxel.
+
+from sklearn.metrics import r2_score
+
+scores = [r2_score(X_train[test], pred, multioutput = 'raw_values')
+         for pred, (train, test) in zip(predictions, cv)]
+
+##############################################################################
+# Mapping the encoding scores on the brain
+# ----------------------------------------
+# To plot the scores onto the brain, we use :func:`nilearn.image.new_img_like`
+# to create a Nifti1Image containing the scores and then threshold it:
+
+from nilearn.image import threshold_img, new_img_like
+from nilearn.masking import unmask
+
+cut_score = np.array(scores).mean(axis=0)
+cut_score[cut_score < 0] = 0
+
+# bring the scores into the shape of the background brain
+score_map = new_img_like(dataset.background,
+                         unmask(cut_score, dataset.mask).get_data())
+
+thresholded_score_map = threshold_img(score_map, threshold = 1e-6)
```

Comment:
We have a convention in nilearn to always end the names of the images with "_img". Would you mind changing  `score_map` to `score_img` and `thresholded_score_map` to `thresholded_score_img` ?